R/junctionCorrection.R

```diff
@@ -198,542 +199,263 @@ updateStrandScoreByRead <- function(unlisted_junction_granges,
-        NA,
-        length(annotatedJunctions)
-    )
-    if (any(mySet.all)) {
-        mySet.training <- (annotatedJunctionsEnd$annotatedEnd.start |
-            annotatedJunctionsEnd$annotatedEnd)[mySet.all]
-        myData <- data.frame(
-            annotatedJunctionsEnd$endScore /
-                (annotatedJunctionsEnd$endScore.start + annotatedJunctionsEnd$endScore),
-            annotatedJunctionsEnd$endScore, annotatedJunctionsEnd$distEnd.start,
-            (annotatedJunctionsEnd$spliceStrand.start == "+"),
-            annotatedJunctionsEnd$spliceStrand.start == "-",
-            (annotatedJunctionsEnd$spliceStrand == "+")
-        )[mySet.all, ]
-        colnames(myData) <- paste("A", seq_len(ncol(myData)), sep = ".")
-        modelmatrix <- model.matrix(~ A.1 + A.2 + A.3 + A.4 + A.5,
-            data = data.frame((myData))
-        ) #+A.6+A.7+A.8+A.9+A.10
-        if (is.null(junctionModel)) {
-            myResults <- fitBinomialModel(
-                labels.train =
-                    as.integer(annotatedJunctionsEnd$annotatedEnd)[mySet.all][mySet.training],
-                data.train = modelmatrix[mySet.training, ], data.test = modelmatrix,
-                show.cv = verbose, maxSize.cv = 10000
-            )
-            junctionModelList[["spliceSitePredictionEnd.start"]] <- myResults[[2]]
-            predictions <- myResults[[1]]
-        }
-        else {
-            predictions <-
-                glmnet:::predict.cv.glmnet(junctionModel[["spliceSitePredictionEnd.start"]],
-                    newx = modelmatrix, s = "lambda.min"
-                )
-        }
-        spliceSitePredictionEnd.start <- rep(NA, length(annotatedJunctionsEnd))
-        names(spliceSitePredictionEnd.start) <- annotatedJunctionsEnd$junctionEndName
-        spliceSitePredictionEnd.start[mySet.all] <- predictions
-        annotatedJunctions$spliceSitePredictionEnd.start <-
-            spliceSitePredictionEnd.start[annotatedJunctions$junctionEndName]
-    }
-    mySet.all <- (annotatedJunctionsEnd$distEnd.end != 0 &
-        annotatedJunctionsEnd$spliceStrand != "*" &
-        annotatedJunctionsEnd$endScore > 0 &
-        abs(annotatedJunctionsEnd$distEnd.end) < 15)
-    annotatedJunctions$spliceSitePredictionEnd.end <- rep(
-        NA,
-        length(annotatedJunctions)
+createSpliceMetadata <- function(annotatedJunctions, splice){
+    metadata <- mcols(annotatedJunctions)[,c(paste0(tolower(splice),'Score'),
+                                            paste0('junction',splice,'Name'),
+                                            paste0('annotated',splice),
+                                            'spliceStrand','spliceMotif')]
+    len <- -length(annotatedJunctions)
+    distdata <- data.frame(
+        dist.start=c(0,diff(start(annotatedJunctions)))*
+        as.integer((seqnames(annotatedJunctions[-1])==seqnames(annotatedJunctions[len]))),
+        annotated.start=c(FALSE,annotatedJunctions$annotatedStart[len]),
```

Comment:
"annotatedStart", shouldn't here be soft coded, instead of hard coded? I believe this will change when the splice side changes? Same for others, please check and make sure they are corresponding to the splice side. You can call it by annotatedJunctions@elementMetadata[,paste0("")] I have examples of this in the testSplice.prime function.

Reply:
These are just temporary names for the columns. The columns are renamed and soft-coded from line 293 to 297.
```
    colnames(distdata) <- 
        c(paste0('dist',splice,'.start'), paste0('annotated',splice,'.start'),
        paste0(tolower(splice),'Score.start'), 'spliceStrand.start','spliceMotif.start',
        paste0('dist',splice,'.end'), paste0('annotated',splice,'.end'),
        paste0(tolower(splice),'Score.end'), 'spliceStrand.end', 'spliceMotif.end')
```